### PR TITLE
feat(Core/PacketIO): correct parsing some opcodes

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- TrinityCore/TrinityCore#18638
- WarheadCore/Warhead@0a2a96e
- WarheadCore/Warhead@b4258ca
- Fix
```
WorldSession::Update ByteBufferException occured while parsing a packet (opcode: 149)
```
## Issues Addressed:
- #6028

## Tests Performed:
-

## How to Test the Changes:
- #6028
- https://youtu.be/CpkJsiTVrFQ
- https://youtu.be/ibUr6XHtE9g

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Server/Protocol/Opcodes.cpp [hunk 1]
```diff
@@ -1396,7 +1396,7 @@ void OpcodeTable::Initialize()
     /*0x4FD*/ DEFINE_SERVER_OPCODE_HANDLER(SMSG_LOOT_SLOT_CHANGED,                                  STATUS_NEVER);
     /*0x4FE*/ DEFINE_HANDLER(UMSG_UPDATE_GROUP_INFO,                                                STATUS_NEVER,      PROCESS_INPLACE,        &WorldSession::Handle_NULL                              );
     /*0x4FF*/ DEFINE_HANDLER(CMSG_READY_FOR_ACCOUNT_DATA_TIMES,                                     STATUS_AUTHED,     PROCESS_THREADUNSAFE,   &WorldSession::HandleReadyForAccountDataTimes           );
-    /*0x500*/ DEFINE_HANDLER(CMSG_QUERY_QUESTS_COMPLETED,                                           STATUS_LOGGEDIN,   PROCESS_THREADSAFE,     &WorldSession::HandleQueryQuestsCompleted               );
+    /*0x500*/ DEFINE_HANDLER(CMSG_QUERY_QUESTS_COMPLETED,                                           STATUS_LOGGEDIN,   PROCESS_INPLACE,        &WorldSession::HandleQueryQuestsCompleted               );
```
Comment (Kitzunu) [inline src/server/game/Server/Protocol/Opcodes.cpp:1399]: What determines if it should be inplace/threadsafe/threadunsafe?

**Comment by Winfidonarleyan** [comment]
https://user-images.githubusercontent.com/18329778/119666121-42fd5000-be5f-11eb-90df-c30c62d4aea9.mp4

File: src/server/game/Server/Protocol/Opcodes.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/server/game/Server/Protocol/Opcodes.cpp:1399]:
```cpp
enum PacketProcessing
{
    PROCESS_INPLACE = 0,                                    //process packet whenever we receive it - mostly for non-handled or non-implemented packets
    PROCESS_THREADUNSAFE,                                   //packet is not thread-safe - process it in World::UpdateSessions()
    PROCESS_THREADSAFE                                      //packet is thread-safe - process it in Map::Update()
};
```

**Comment by tkn963** [comment]
@Winfidonarleyan If it works the same way for Healing over Time (Renew etc) and overheal - which I'm guessing does since it's basically all just incoming heals, then yes. I can make a video of the system when it worked to show more specifically but I would assume that what you have is what I'm refering to.

**Comment by tkn963** [comment]
Confirmed! And a thousand thumbs up for the fix

**Comment by Winfidonarleyan** [comment]
Unfortunately, I broke it :/

> Confirmed! And a thousand thumbs up for the fix

**Comment by Winfidonarleyan** [comment]
> Confirmed! And a thousand thumbs up for the fix

Very good, thanks for testing

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**